ddsrouter_core/src/cpp/dynamic/DiscoveryDatabase.hpp

```diff
@@ -52,14 +81,58 @@ class DiscoveryDatabase
     bool endpoint_exists(
             const types::Guid& guid) const noexcept;
 
+    /**
+     * @brief Add a new endpoint to the database.
+     *
+     * @param [in] new_endpoint: new endpoint to store
+     */
+    void add_endpoint(
```

Comment:
As the current behaviour is not this, but push the new data to a queue to be processed in the future, I would update the comment.
The same for the other methods below.